sqlglot/optimizer/qualify_columns.py

```diff
@@ -898,10 +898,21 @@ def get_source_columns(self, name: str, only_visible: bool = False) -> t.Sequenc
                     for (name, alias) in itertools.zip_longest(columns, column_aliases)
                 ]
 
+            pseudocolumns = self._get_source_pseudocolumns(name)
+            if pseudocolumns:
+                columns = list(columns) + list(pseudocolumns)
```

Comment:
Due to the `t.Sequence` constraint on these values we can't just apply `+` as if we had lists.